app/controllers/posts_controller.rb

```diff
@@ -1,23 +1,26 @@
 class PostsController < ApplicationController
-  before_action :set_user
-  before_action :set_post, only: [:show]
-
   def index
-    @posts = Post.all
-    render 'index'
+    @user = User.find(params[:user_id])
+    @posts = @user.posts
   end
 
   def show
-    render 'show'
+    @post = Post.find(params[:id])
   end
 
-  private
-
-  def set_user
-    @user = User.find(params[:user_id])
+  def new
+    @post = Post.new
   end
 
-  def set_post
-    @post = @user.posts.find(params[:id])
+  def create
+    @post = current_user.posts.build(params.require(:post).permit(:title, :text))
+    if @post.save
+      flash[:success] = 'Post saved successfully'
+      redirect_to user_posts_path
+    else
+      flash.now[:error] = 'Post not saved, try again later'
+      puts puts @post.errors.full_messages
+      render :new
+    end
   end
```

Comment:
- [ Optional ] Good Job for A post is created on behalf of the currently logged in user (current_user) of the app, But Kindly I recommend you to use `the strong parameters`  with the below params method , Please check out this [link](`https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/lessons/Controllers_lesson.md#:~:text=Pay%20special%20attention%20to%20the%20strong%20parameters%20description%20in%20Rails%20Guides.`) for more info.

```ruby
def create
    @post = current_user.posts.build(post_params)
    if @post.save
      flash[:success] = 'Post saved successfully'
      redirect_to user_posts_path
    else
      flash.now[:error] = 'Post not saved, try again later'
      puts puts @post.errors.full_messages
      render :new
    end

    
    private 
    def post_params
      params.require(:post).permit(:title, :text)
    end
```